src/components/AllProducts.js

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect } from 'react';
+import { getAllProducts } from '../api';
+import Product from './Product.js';
+import './AllProducts.css'
+
+const AllProducts = (props) => {
+    const {allProducts, setAllProducts} = props;
+    useEffect (() => {
```

Comment:
Instead, create a `getProducts` function, that does all of this.  That way, we can just call `getProducts()`, and it fetches and sets products on state.